rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,233 @@
+      value={value} 
+      onChange={handleOnChange} 
+      defaultValue={defaultValue} 
+      ref={inputRef}
+    />
+  )
+};
+
+export default MyComponent;
+```
+## Component Example - Controlled
+
+Here is an example of how a basic component will look in JS without uncontrolled support.
+
+```jsx 
+// my-component.component.js
+const MyComponent = ({ value, onChange, ...rest }) => {
+   const handleOnChange = (event) => {
+    if (onChange) {
+      onChange(event);
+    } 
+  };
+
+  /* Component logic goes here */
+
+  return <input value={value} onChange={handleOnChange} />;
+};
+
+MyComponent.propTypes = {
+  value: PropTypes.string.isRequired,
+  onChange: PropTypes.function.isRequired,
+};
+
+export default MyComponent;
+```
+Here is an example of how a basic component will look in TS without uncontrolled support.
+
+```tsx
+// my-component.component.tsx
+interface MyComponentProps = {
+  value: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+const MyComponent = ({ value, onChange, ...rest }: myComponentProps) => {
+   const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (onChange) {
+      onChange(event.target.value);
+    } 
+  };
+
+  /* Component logic goes here */
+
+  return <input value={value} onChange={handleOnChange} />;
+};
+
+export default MyComponent;
+```
+# Detailed design
+
+- We will add deprecation warnings to our input components so that our consumers are aware of the pending changes to remove uncontrolled support. This will appear as a console warning if our consumer is using the uncontrolled mode and a warning on our Storybook documentation.
+
+- We will remove uncontrolled behaviour from our input components and associated props. This will be done in one Pull Request with each component having it's own commit message with a `BREAKING CHANGE`. The benefit of this is that we will only release one major version of Carbon. 
+
+This approach will provide immediate benefit to developers of Carbon as it means we will have less code and tests to maintain. It will also mean that consumers of Carbon can steadily upgrade their version of Carbon.
+
+## Components
+
+The input components that currently support being uncontrolled are: 
+1. Textbox
+1. Decimal
+1. Number Input
+1. Simple Select
+1. Filterable Select
+1. Multi Select
+1. Grouped Character
+1. Numeral Date
+1. Search
+1. Checkbox 
+1. RadioButton 
+1. Switch 
+1. SimpleColorPicker 
+1. AdvancedColorPicker 
+1. ButtonToggle
+
+## Unit Testing
+
+Currently uncontrolled and controlled behaviours are unit tested. As part of this work we could reduce the amount of code in the spec files quite considerably. If we take `Decimal` as an example of this, `Decimal` currently has 1430 lines of code in the spec file. This could be reduced to down to around 480 lines of code if we were to remove the uncontrolled tests. This would make the tests easier to read, maintain and reduce the runtime of the tests too.
```

Comment:
similar to my first comment above, I would add the word "both" - `... are both unit tested` - as I think it's clearer. Or perhaps better still, rewrite the sentence as `Currently we have unit tests for both controlled and uncontrolled behaviours.`

Also the last sentence is not quite grammatical - we're not making it "easier to reduce the runtime", we're actually reducing the runtime. I think the phrasing should be `This would make the tests easier to read and maintain, while also reducing the runtime of the tests.` ("while also reducing" could be "and also reduce", or if you prefer to keep the "too" you can have that at the end and remove "also". Many options!)

Reply:
Valid point. I'll get this changed 😄